search.go

```diff
@@ -0,0 +1,34 @@
+package registry
+
+import (
+	"fmt"
+)
+
+// SearchParams encapsulates parameters provided to Searchable.Search
+type SearchParams struct {
+	Q             string
+	Limit, Offset int
+}
+
+// Searchable is an interface for supporting search
+type Searchable interface {
+	Search(p SearchParams) ([]Result, error)
+}
+
+// ErrSearchNotSupported is returned for collections that are not searchable
```

Comment:
I try to avoid language like "is returned", instead I'd say something like:
```golang
// ErrSearchNotSupported is the canonical error to indicate search isn't implemented
```
by not referring to the error as a return value it allows the error to be considered with less accompanying context.